desktop/ui/src/main/java/org/datacleaner/widgets/HdfsUrlTextFieldImpl.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.widgets;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import org.apache.metamodel.util.HdfsResource;
+import org.datacleaner.windows.HdfsUrlChooser;
+
+public class HdfsUrlTextFieldImpl extends AbstractFilenameTextField<HdfsResource> {
```

Comment:
Hmmm... Hmmmm... I still don't really agree. It's still not really a file (at least not in the Java sense of the word).We could call it AbstractResourceTextField, with HdfsResourceTextField, and FileResourceTextField.

... Of course, that will mean that we lose some of the actual meaning (it can no longer be inferred that it is the actual Url and name that it is being used for). HdfsUrlResourceTextField and FilenameResourceTextField, maybe?